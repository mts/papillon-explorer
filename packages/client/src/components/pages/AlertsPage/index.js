import React from 'react';
import { Link } from 'react-router-dom';

const AlertsPage = () => {
  return (
    <div>
      <h2>Alerts Page</h2>
      <div>
        Under construction ðŸ™‚
        <Link to="/">Back to Home Page</Link>
      </div>
    </div>
  )
}

export default AlertsPage;
