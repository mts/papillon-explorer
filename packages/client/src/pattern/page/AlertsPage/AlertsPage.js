import React from 'react'
import { Query } from 'react-apollo'
import { startPageRoute } from '../../../routes'
import { PAGES } from '../../../graphql/query'
import AlertsTemplate from '../../template/AlertsTemplate'

const AlertsPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? (
          <AlertsTemplate page={data.pages.pages.find(page => page.id === 'alerts')} startPageRoute={startPageRoute} />
        ) : null
      }
    </Query>
  )
}

export default AlertsPage
