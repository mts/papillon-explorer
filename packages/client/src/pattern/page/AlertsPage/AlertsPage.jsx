import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import { PAGES } from '../../../graphql/query'
import { AlertsTemplate } from '../../template/AlertsTemplate'

export const AlertsPage = () => {
  const { data } = useQuery(PAGES)

  if (data?.pages?.pages) {
    return <AlertsTemplate page={data.pages.pages.find(page => page.id === 'alerts')} />
  }

  return null
}

export default AlertsPage
