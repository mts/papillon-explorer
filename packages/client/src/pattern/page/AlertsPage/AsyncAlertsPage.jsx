import React from 'react'

const LazyAlertsPage = React.lazy(() => import('./AlertsPage'))

export const AsyncAlertsPage = () => (
  <React.Suspense fallback={null}>
    <LazyAlertsPage />
  </React.Suspense>
)
