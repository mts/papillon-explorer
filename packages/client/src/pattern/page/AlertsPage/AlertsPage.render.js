import React from 'react'
import { AsyncAlertsPage } from './AsyncAlertsPage'
import { AlertsPage } from './AlertsPage'

export const alertsPageAsync = <AsyncAlertsPage />
export const alertsPageDefault = <AlertsPage />
