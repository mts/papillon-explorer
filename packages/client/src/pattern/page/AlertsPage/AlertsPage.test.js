import React from 'react'
import { alertsPageAsync, alertsPageDefault } from './AlertsPage.render'

jest.mock('@apollo/react-hooks', () => ({ useQuery: () => {} }))
jest.mock('react-router-dom', () => ({ Link: () => <div>some link</div> }))

const apolloReactHooksMockObject = require('@apollo/react-hooks')

describe('<AlertsPage />', () => {
  const pagesDataObject = {
    data: {
      pages: {
        pages: [{ id: 'alerts', name: 'some-name' }],
      },
    },
  }

  afterEach(() => jest.clearAllMocks())

  describe('Snaphot', () => {
    test('must match alertsPageAsync', () => {
      jest.spyOn(apolloReactHooksMockObject, 'useQuery').mockReturnValue(pagesDataObject)
      expect(global.renderToJSON(alertsPageAsync)).toMatchSnapshot()
    })

    test('must match alertsPageDefault', () => {
      expect(global.renderToJSON(alertsPageDefault)).toMatchSnapshot()
    })

    test('must match alertsPageNull', () => {
      jest.spyOn(apolloReactHooksMockObject, 'useQuery').mockReturnValue({})
      expect(global.renderToJSON(alertsPageDefault)).toMatchSnapshot()
    })
  })
})
