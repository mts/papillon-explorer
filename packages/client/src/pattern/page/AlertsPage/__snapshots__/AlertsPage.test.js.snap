// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertsPage /> Snaphot must match alertsPageAsync 1`] = `null`;

exports[`<AlertsPage /> Snaphot must match alertsPageDefault 1`] = `
<div>
  <h2>
    some-name
  </h2>
  Under construction
   
  <span
    aria-label="emoji"
    role="img"
  >
    ðŸ™‚
  </span>
  <div>
    some link
  </div>
</div>
`;

exports[`<AlertsPage /> Snaphot must match alertsPageNull 1`] = `null`;
